model/project_parser.go

```diff
@@ -928,6 +920,50 @@ func evaluateBVTasks(tse *taskSelectorEvaluator, tgse *tagSelectorEvaluator, vse
 	return ts, evalErrs
 }
 
+// getParserBuildVariantTaskUnit combines the parser project's task definition with the build variant task definition.
+// DependsOn will be handled separately.
+func getParserBuildVariantTaskUnit(name string, pt parserTask, bvt parserBVTaskUnit) BuildVariantTaskUnit {
+	res := BuildVariantTaskUnit{
+		Name:             name,
+		Patchable:        bvt.Patchable,
+		PatchOnly:        bvt.PatchOnly,
+		AllowForGitTag:   bvt.AllowForGitTag,
+		GitTagOnly:       bvt.GitTagOnly,
+		Priority:         bvt.Priority,
+		ExecTimeoutSecs:  bvt.ExecTimeoutSecs,
+		Stepback:         bvt.Stepback,
+		Distros:          bvt.Distros,
+		CommitQueueMerge: bvt.CommitQueueMerge,
+		CronBatchTime:    bvt.CronBatchTime,
+		BatchTime:        bvt.BatchTime,
+	}
+	if res.Priority == 0 {
+		res.Priority = pt.Priority
+	}
+	if res.Patchable == nil {
+		res.Patchable = pt.Patchable
+	}
+	if res.PatchOnly == nil {
+		res.PatchOnly = pt.PatchOnly
+	}
+	if res.AllowForGitTag == nil {
+		res.AllowForGitTag = pt.AllowForGitTag
+	}
+	if res.GitTagOnly == nil {
+		res.GitTagOnly = pt.GitTagOnly
+	}
+	if res.ExecTimeoutSecs == 0 {
+		res.ExecTimeoutSecs = pt.ExecTimeoutSecs
+	}
+	if res.Stepback == nil {
+		res.Stepback = pt.Stepback
+	}
+	if len(res.Distros) == 0 {
```

Comment:
Should this also check RunOn? It looks like both fields are supposed to be identical

Reply:
Distros is set to RunOn during unmarshallling https://github.com/evergreen-ci/evergreen/blob/35d4ddcae4b8b9010ba0b5140b0e7ebb251b33ce/model/project_parser.go#L415

Reply:
Sorry I was referring to that part actually. If you set distros but not RunOn, we don't do the copy. Will that cause problems?

Reply:
I'm actually going to switch this copy logic; since Distros is what we want to depreciate, that's the one we should ignore.